test/e2e/operator.go

```diff
@@ -272,3 +277,99 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var location, resourceGroup, subnet, vnet string
+	var correctNSG *string
+	var dummynsg mgmtnetwork.SecurityGroup
+	type subnetDescriptor struct {
+		resourceGroup string
+		vnetName      string
+		subnetName    string
+		location      string
+	}
+	BeforeEach(func() {
+		enableReconcileNSG := func() {
```

Comment:
Overall bit strange pattern to define functions like this here and after this just call them once. 
Either just do all this in the before or define functions outside and call them. confusing to read.

Reply:
moved the funcs above BeforeEach. I added in the funcs to try and break it down so it was more clear the different steps/tasks I was doing.